### DISCUSSION TITLE
Split image into 128x128 chunks using +repage and number the resulting files based on the x/y position of the file.

### DISCUSSION THREAD

**Discussion by IoIxD** [ImageMagick/ImageMagick#7086] [Help] [answered]
I have a 21600x10800 heightmap of the Earth from NASA. I want to split it into a bunch of 128x128 chunks, *using +repage and not a script of my own*, and then number each file based on the x/y position of the chunk in the original image. Can I do this with ImageMagick?

**Comment by snibgo** [comment, marked as answer]
Don't use `+repage`. Use something like this:
```
magick ^
  input.png ^
  -crop 128x128 ^
  -set filename:MYFILE %%O ^
  myprefix_%%[filename:MYFILE].png
```
That is Windows BAT syntax. For bash, change ^ to \\, and %% to %.

Output filenames will have names like `myprefix_+140+220.png`.

**Reply by Buggem** [reply]
Bash version here for convenience
```
magick \
  input.png \
  -crop 128x128 \
  -set filename:MYFILE %O \
  myprefix_%[filename:MYFILE].png
```

**Comment by fmw42** [comment]
To the OP. When asking a question, please identify your IM version and your platform/OS
